src/frame_graph/frame_graph.hpp

```diff
@@ -762,19 +761,17 @@ namespace wr
 			{
 				if (typeid(T) == m_data_type_info[i])
 				{
-					try {
-						return std::any_cast<R>(m_settings[i].value());
-					}
-					catch (const std::bad_any_cast & e) {
-						LOGW("A task settings requested failed to cast to T. ({})", e.what());
-						return R();
-					}
+					return std::any_cast<R>(m_settings[i].value());
 				}
 			}
 
 			LOGC("Failed to find task settings! Does your frame graph contain this task?");
 			return R();
 		}
+		catch (const std::bad_any_cast& e) {
+			LOGW("A task settings requested failed to cast to T. ({})", e.what());
```

Comment:
when settings cannot be found it logs a critical. if it is a bad cast it is only a warning. how severe is it when this operation fails?